Signal/src/AppDelegate.m

```diff
@@ -197,9 +197,10 @@ - (UIViewController *)loadingRootViewController
         [[UIStoryboard storyboardWithName:@"Launch Screen" bundle:nil] instantiateInitialViewController];
 
     BOOL shouldShowUpgradeLabel = NO;
-    NSString *previousVersion = AppVersion.instance.lastAppVersion;
+    NSString *previousVersion = AppVersion.instance.lastCompletedLaunchAppVersion;
     // We added a number of database views in v2.13.0.
-    if ([VersionMigrations isVersion:previousVersion atLeast:@"2.0.0" andLessThan:@"2.13.0"]) {
+    if ([TSAccountManager isRegistered] &&
```

Comment:
> If we just omit it, people who have previously launched, but never completed registration, then upgraded signal, then decide to complete registration, will see a short blip of the upgrading screen.

I've verified that this does block during the database upgrade process.  Additionally, there's many other places we're doing DB reads in `appDidFinishLaunching...`, not least in `[VersionMigrations performUpdateCheck]` which calls this exact method `[TSAccountManager isRegistered]` three times. 

So I'm going to leave this in.